demos/demos_databases_apis/microsoft/sentinel/sentinel_security_analysis.ipynb

```diff
@@ -0,0 +1,529 @@
+    "| where TimeGenerated > ago(30d)\n",
+    "| project IncidentNumber, Title, Severity, Status, Owner;\n",
+    "\n",
+    "// Get related alerts  \n",
+    "let alerts = SecurityAlert\n",
+    "| where TimeGenerated > ago(30d)\n",
+    "| project AlertName, CompromisedEntity, AlertSeverity, Tactics;\n",
+    "\n",
+    "// Join and analyze\n",
+    "incidents\n",
+    "| join kind=inner (alerts) on $left.Title == $right.AlertName\n",
+    "| summarize \n",
+    "    IncidentCount = dcount(IncidentNumber),\n",
+    "    AffectedEntities = dcount(CompromisedEntity),\n",
+    "    TacticsUsed = make_set(Tactics)\n",
+    "    by Title, Severity\n",
+    "| order by IncidentCount desc\n",
+    "\"\"\"\n",
+    "\n",
+    "try:\n",
+    "    correlation_data = g.kql(correlation_query, timespan=timedelta(days=30))\n",
+    "    print(f\"Found {len(correlation_data)} incident-alert correlations\")\n",
+    "    if len(correlation_data) > 0:\n",
+    "        print(correlation_data.head())\n",
+    "except Exception as e:\n",
+    "    print(f\"Correlation query failed: {e}\")\n",
+    "    print(\"This requires both SecurityIncident and SecurityAlert tables\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": "## Summary\n\nThis notebook demonstrated:\n\n1. **Connecting to Microsoft Sentinel** using Azure authentication (device code, service principal, or DefaultAzureCredential)\n2. **Exploring available data** with `sentinel_tables()` and `sentinel_schema()`\n3. **Security analysis** using KQL queries for:\n   - Failed login analysis\n   - Security alerts monitoring\n   - Network traffic analysis\n4. **Graph visualization** of:\n   - User-IP relationships\n   - Alert correlations\n5. **Advanced correlation** across multiple data sources\n\n## Next Steps\n\n- **Customize queries** for your specific security use cases and available data tables\n- **Create automated dashboards** by scheduling notebook execution\n- **Integrate with threat intelligence** feeds using additional KQL joins\n- **Build detection rules** based on graph patterns you discover\n- **Scale analysis** by adjusting time windows and data volumes\n\n## Troubleshooting Tips\n\n- **No data found**: Some workspaces may not have SecurityEvent, SigninLogs, or SecurityAlert tables\n- **Authentication issues**: Try `az login` first, or check your service principal credentials\n- **Permission errors**: Ensure your account has Log Analytics Reader permissions\n- **Empty results**: Adjust time ranges - some workspaces have limited data retention\n\n## Resources\n\n- [Microsoft Sentinel KQL Reference](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/)\n- [Graphistry Documentation](https://pygraphistry.readthedocs.io/)\n- [Azure Monitor Query Documentation](https://docs.microsoft.com/en-us/python/api/azure-monitor-query/)\n- [Sentinel Data Connectors](https://docs.microsoft.com/en-us/azure/sentinel/connect-data-sources)"
```

Comment:
Next steps should include links, especially for most recommended

* Try for free with your graphistry hub account and jupyter / google colab notebooks
* Check out louie.ai for helping generating these queries and visualizations through genai
* Check out the Kusto tutorial 
* Learn more about complementary pygraphistry  features like UMAP for detecting patterns & outliers on events, GFQL for mining graphs once you've made them, and how to control graph colors and layouts

etc